File: my-folder/0300-longest-increasing-subsequence/solution.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        freqCounter=[1]*len(nums)
        for i in range(1,len(nums)):
            for j in range(i): 
                if nums[j] < nums[i]:
                    freqCounter[i] = max(1+freqCounter[j], freqCounter[i]) 
        print(freqCounter)
        return max(freqCounter)
                

                

        

File: my-folder/0300-longest-increasing-subsequence/solution.java
public class Solution {
    public int lengthOfLIS(int[] nums) {
        int[] LIS = new int[nums.length];
        Arrays.fill(LIS, 1);

        for (int i = nums.length - 1; i >= 0; i--) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] < nums[j]) {
                    LIS[i] = Math.max(LIS[i], 1 + LIS[j]);
                }
            }
        }
        return Arrays.stream(LIS).max().getAsInt();
    }
}
// dp[4]
// 3 
// max(1, 3) = 3
// max(3, 1)
//   7 101 18
// dp[3]
// 5,
//   3,7,101,18
// dp[3] = 1 + dp[5] = 4
//       = 1 + dp[6] = 4 its max

// dp = [2 2 4  3 3 2 1 1]
// max(dp) = 4

